src/Data/Conduit/Tar/Windows.hs

```diff
@@ -67,6 +68,7 @@ restoreFileInternal lenient fi@FileInfo {..} = do
                                     (`when` Dir.setModificationTime fpStr modTime))
                 return (fi, either ((excs ++) . pure) (const excs) eExc)
         FTNormal -> do
+            Dir.createDirectoryIfMissing True $ FilePath.takeDirectory fpStr
```

Comment:
Missing `liftIO` as well as `when lenient` conditional, just as in `Unix.hs` module.